docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,271 @@
+  }
+  ```
+  <template v-slot:caption>
+    .eslintrc.json
+  </template>
+</AposCodeBlock>
+
+When you call `npm test`, it will lint your files according to [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe) rules and run the tests using [mocha](https://www.npmjs.com/package/mocha).
+
+### Test folder
+
+You will need a `test/package.json` file referencing the repository URL of your module. Please replace `article` & `%article-repository-url%` with your module informations.
+
+e.g. for the module [@apostrophecms/login-hcaptcha](https://github.com/apostrophecms/login-hcaptcha) we use 
+
+> `"@apostrophecms/login-hcaptcha": "git://github.com/apostrophecms/login-hcaptcha.git"`
+> 
+
+<AposCodeBlock>
+  ```json
+  {
+    "/**": "This package.json file is not actually installed.",
+    " * ": "Apostrophe requires that all npm modules to be loaded by moog",
+    " */": "exist in package.json at project level, which for a test is here",
+    "dependencies": {
+      "apostrophe": "^3.26.0",
+      "article": "%article-repository-url%"
+    }
+  }
+  ```
+  <template v-slot:caption>
+    test/package.json
+  </template>
+</AposCodeBlock>
+
+You don’t want to commit test files artifacts to git, please add the following to your `.gitignore` file
+
+<AposCodeBlock>
+  ```gitignore
+  # Test files
+  /test/apos-build
+  /test/public
+  /test/locales
+  ```
+  <template v-slot:caption>
+    .gitignore
+  </template>
+</AposCodeBlock>
+
+## Writing tests
+
+We will start with a single test file `test/index.js`.
+
+As your test grows, you can break them down into multiple files based on your liking.
+
+Mocha does not play well with arrow-functions, more info at [https://mochajs.org/#arrow-functions](https://mochajs.org/#arrow-functions)
+
+<AposCodeBlock>
+  ```javascript
+  // We use assert strict assertion mode, value and type must be the same when doing a comparison, .i.e. ===
+  // https://nodejs.org/api/assert.html#strict-assertion-mode
+  const assert = require('assert').strict;
+  const t = require('apostrophe/test-lib/util.js');
+
+  // getAppConfig is used to set options for Apostrophe and the module you want to test
+  const getAppConfig = function (options = {}) {
+    return {
+      article: {
+        options: {
+          // Pass the required options here
+          // In the example below, we will test for the presence of a `custom` option
+          // If your module doesn't have any options, please skip the `options` attribute
+          ...options,
+          custom: true
+        }
+      }
+    };
+  };
+
+  // describe and it functions are provided by mocha
+  // Usually use to "structure" the test file
+  // They don't have a strict meaning from mocha's point of view
+
+  // describe is a test suite and used to group tests
+  // You can have as many describe as you want in a test file
+  // describe can be nested
+  describe('article', function () {
+    let apos;
+
+    this.timeout(t.timeout);
```

Comment:
Can you add a comment about whether this is optional and that you need to set an environment variable? I know you have a description of setting it below, but a reminder here would be good.